### PR TITLE
Toolchain: Update to LLVM 15.0.0

### PR DESCRIPTION
**Pull request by timschumi**
This updates LLVM (both the toolchain and the port) to 15.0.0, and fixes or supresses a few discovered "issues" along the way.

One remaining problem is that the following messages get generated while linking the kernel:

```
ld.lld: warning: section type mismatch for .ro_after_init
>>> Kernel/CMakeFiles/Kernel.dir/Bus/PCI/Initializer.cpp.o:(.ro_after_init): SHT_PROGBITS
>>> output section .ro_after_init: SHT_NOBITS

[...]

ld.lld: warning: section type mismatch for .heap
>>> Kernel/libkernel_heap.a(kmalloc.cpp.o):(.heap): SHT_PROGBITS
>>> output section .bss: SHT_NOBITS
```

This doesn't appear to affect anything in practice (presumably because we just ignore those flags while mapping the kernel?), but it's still a bit noisy while building.

To save on toolchain rebuilds, please make sure that #15149 is reviewed and submittable before merging this, and merge both in tandem.

### CODE REVIEW

**Comment by timschumi** [comment]
Also, from the "toolchain update checklist" (#14347) we still have the following:

- Figure out if "[FIXME: Implement -static-libstdc++ in the next toolchain update](https://github.com/SerenityOS/serenity/blob/c6eaa39b990e2c654e09cc3bf7605de7ec7c13be/Meta/CMake/utils.cmake#L71)" still applies.
- Remove [Clang compatibility symlinks that have been duplicated into `Ports`](https://github.com/SerenityOS/serenity/blob/c6eaa39b990e2c654e09cc3bf7605de7ec7c13be/Ports/.hosted_defs.sh#L36-L41).
- Remove `_aligned_alloc` and `_aligned_free` from LibC and stdlib.h

The second and last I can handle tomorrow, the first I'm unsure about, but I guess we can just try if it works.

File: Userland/Libraries/LibRegex/RegexMatcher.cpp [hunk 1]
```diff
@@ -415,7 +415,6 @@ template<class Parser>
 bool Matcher<Parser>::execute(MatchInput const& input, MatchState& state, size_t& operations) const
 {
     BumpAllocatedLinkedList<MatchState> states_to_try_next;
-    size_t recursion_level = 0;
```
Comment (bgianfo) [inline Userland/Libraries/LibRegex/RegexMatcher.cpp:418]: You you missed the usage in under `#if REGEX_DEBUG` below.

Comment (timschumi) [reply Userland/Libraries/LibRegex/RegexMatcher.cpp:418]: "Curse you, `ALL_DEBUG`!"

Should be fixed now.

**Comment by timschumi** [comment]
cc @ADKaster, as I'm not knowledgeable enough about the coverage implementation:

Linking in the `.profile` variant of `clang-rt` is usually handled by `-static-libstdc++` internally, and the special case for `ENABLE_USERSPACE_COVERAGE_COLLECTION` was just a duplication of logic, right?

**Review by BenWiederhake — APPROVED** [review]
Looks good, and is easy to read. Great! :D

**Comment by ADKaster** [comment]
> the special case for ENABLE_USERSPACE_COVERAGE_COLLECTION was just a duplication of logic, right?

Yes, it didn't link otherwise. If it links now we're good to go

**Comment by Hendiadyoin1** [comment]
Can this also resolve this comment in Kernels CMakeLists:
[`# FIXME: remove -nodefaultlibs after the next toolchain update`](https://github.com/SerenityOS/serenity/blob/a60c5166c69d6e3e423019706c9d1b722a9b15e0/Kernel/CMakeLists.txt#L527)

**Comment by timschumi** [comment]
> Can this also resolve this comment in Kernels CMakeLists: [`# FIXME: remove -nodefaultlibs after the next toolchain update`](https://github.com/SerenityOS/serenity/blob/a60c5166c69d6e3e423019706c9d1b722a9b15e0/Kernel/CMakeLists.txt#L527)

Maybe, but that appears to be GCC-specific, and resolved ages ago anyways. I can look into it, but I don't think there is a reason to not submit it immediately and independently of LLVM 15.

**Comment by bgianfo** [comment]
Pulling the trigger while Europe is waking up / asleep :)

**Pull request merged**